lib/Cron/Workers/SubscribersCountCacheRecalculation.php

```diff
@@ -0,0 +1,75 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Cron\Workers;
+
+use MailPoet\Cache\TransientCache;
+use MailPoet\Entities\SegmentEntity;
+use MailPoet\Models\ScheduledTask;
+use MailPoet\Segments\SegmentsRepository;
+use MailPoet\Segments\SegmentSubscribersRepository;
+use MailPoet\WP\Functions as WPFunctions;
+use MailPoetVendor\Carbon\Carbon;
+
+class SubscribersCountCacheRecalculation extends SimpleWorker {
+  private const EXPIRATION_IN_MINUTES = 30;
+  const TASK_TYPE = 'subscribers_count_cache_recalculation';
+  const AUTOMATIC_SCHEDULING = false;
+  const SUPPORT_MULTIPLE_INSTANCES = false;
+
+  /** @var TransientCache */
+  private $transientCache;
+
+  /** @var SegmentsRepository */
+  private $segmentsRepository;
+
+  /** @var SegmentSubscribersRepository */
+  private $segmentSubscribersRepository;
+
+  public function __construct(
+    TransientCache $transientCache,
+    SegmentsRepository $segmentsRepository,
+    SegmentSubscribersRepository $segmentSubscribersRepository,
+    WPFunctions $wp
+  ) {
+    parent::__construct($wp);
+    $this->transientCache = $transientCache;
+    $this->segmentsRepository = $segmentsRepository;
+    $this->segmentSubscribersRepository = $segmentSubscribersRepository;
+  }
+
+  public function processTaskStrategy(ScheduledTask $task, $timer) {
+    $segments = $this->segmentsRepository->findAll();
```

Comment:
I am not sure how this works if this calculation goes always in the same order. Would the first segment be always up to date but the last one may fail to calculate because of the execution limit?

Reply:
This problem should solve the check on line 56. When the cached item is up to date, it is skipped.